pkg/util/pem.go

```diff
@@ -121,3 +85,33 @@ func DecodeX509CertificateChainBytes(certBytes []byte) ([]*x509.Certificate, err
 
 	return certs, nil
 }
+
+// Get the split bundle of all certificates in the certificates pool as representation of [][]byte
+func AsSplitPEMBundle(certPool *CertPool) [][]byte {
+	return certPool.getCertsPEM()
+}
+
+// Get the split bundle of all certificates in the certificates pool as representation of []byte
+func AsPEMBundleBytes(certPool *CertPool) []byte {
+	return bytes.TrimSpace(bytes.Join(certPool.getCertsPEM(), nil))
+}
+
+// Get the split bundle of all certificates in the certificates pool as representation of []string
+func AsPEMBundleStrings(certPool *CertPool) []string {
+	var certList = make([]string, 0)
+
+	for _, cert := range certPool.getCertsPEM() {
+		certList = append(certList, strings.TrimSpace(string(cert)))
+	}
+
+	return certList
+}
+
+// Get the list of all x509 Certificates in the certificates pool
+func AsCertificateList(certPool *CertPool) []*x509.Certificate {
+	return certPool.getCertsList()
+}
+
+func GetCertificatesQuantity(certPool *CertPool) int {
+	return certPool.size()
+}
```

Comment:
Why can't we just export the `size` function and delete this function?

Reply:
It would be easy to calculate a size for other kind of storages in the future. As was stated earlier it was done because I wanted to encapsulate the insides of `Certificates pool`.

Reply:
Ok, I don't understand your argument, but this can stay - it's "just" a name. 😉 If you still want to alias the `Size` function I think `GetCertificateCount` could be a better name.

Reply:
> I wanted to encapsulate the insides of Certificates pool.

Thanks for the clarification.

Can you elaborate on the need for encapsulating CertPool's behavior, and why the count of certificate in the certificate pool shouldn't be exposed? I don't think that the behaviors "counting certificates", "exporting certificates to a PEM bundle", and "exporting certificates to a slide of DER bytes" need to be encapsulated away. More generally, I'm not convinced that encapsulation will solve anything in this context; I'd love to know more, though!

Reply:
Behavior of `CertPool` should be hidden from its consumers. For example we are going to substitute current `CertPool` by one from `x509` package or even third-party. As little possibilities to works with `CertPool` directly are given to consumers, as little components coupling we have. 
From my opinion `CertPool` acts like `io.Writer` or `io.Reader`.

Reply:
> Behavior of `CertPool` should be hidden from its consumers. For example we are going to substitute current `CertPool` by one from `x509` package or even third-party. As little possibilities to works with `CertPool` directly are given to consumers, as little components coupling we have. From my opinion `CertPool` acts like `io.Writer` or `io.Reader`.

This does not make sense to me. Creating util-wrapper functions does not change your stated concern. It just looks very strange to me to have dummy single-statement functions. Making the `CertPool` type internal probably makes sense - to allow ourselves more freedom to modify it over time.